src/actions/shared/approve.ts

```diff
@@ -0,0 +1,31 @@
+import { Token, TOKEN_PROGRAM_ID, u64 } from '@solana/spl-token';
+import { Keypair, PublicKey, Transaction } from '@solana/web3.js';
+
+interface CreateApproveParams {
+  authority?: Keypair;
+  account: PublicKey;
+  owner: PublicKey;
+  amount: number | u64;
+}
+
+export function createApproveTxs({
+  authority = Keypair.generate(),
+  account,
+  owner,
+  amount,
+}: CreateApproveParams) {
+  const createApproveTx = new Transaction().add(
+    Token.createApproveInstruction(
+      TOKEN_PROGRAM_ID,
+      account,
+      authority.publicKey,
```

Comment:
We don't have `strict: true` in our `tsconfig` yet, but from purely types this could be `null` which points to the types not checking out since the actual code shows that it will never be `null`.
So once we turn this on (which we should eventually) you'll most likely get a type error.

I'd like to see the fact that the passed params have optional `authority` but are set to a generated `Keypair` if not provided in the types.

Essentially the args the function receives (after providing the default for `authority`) are something akin to `CreateApproveParams & { authority: Keypair }` removing the `?`.

This is best expressed via an optional property type as explained in [this answer](https://stackoverflow.com/a/61108377/97443) and the one below.

Reply:
To be clear, you'd remove the `?` from the typedef and wrap the params type in an `Optional< CreateApproveParams, 'authority'>`

Reply:
So something akin to this in final:

```ts
export function createApproveTxs(args: Optional<CreateApproveParams, 'autority'>) {
  const {
    authority = Keypair.generate(),
    account,
    owner,
    amount,
  }: CreateApproveParams = args
  [..]
}
```